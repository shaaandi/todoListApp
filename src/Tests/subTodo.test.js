import React from 'react';
import {shallow} from 'enzyme';
import SubTodo from '../components/jsfiles/subTodo';



it('SubTodo Testing', () => {
    let dummyData = {
        title : 'Drink Milk',
        state : false,
        id : 'fsgd45trgf',
        parentKey : 'dsfgfdgre4esfdz',
        prevTodo : {}
    }
    let wrapper = shallow(<SubTodo {...dummyData}/>);
    expect(wrapper).toMatchSnapshot();
})