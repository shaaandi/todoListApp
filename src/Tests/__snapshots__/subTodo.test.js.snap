// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTodo Testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubTodo
    id="fsgd45trgf"
    parentKey="dsfgfdgre4esfdz"
    prevTodo={Object {}}
    state={false}
    title="Drink Milk"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="stateButton"
          onClick={[Function]}
        >
          <FaCircleNotch
            size={18}
          />
        </button>,
        <div
          className="subTodo-card"
        >
          <div>
            Drink Milk
          </div>
        </div>,
      ],
      "className": "subTodo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FaCircleNotch
            size={18}
          />,
          "className": "stateButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            Drink Milk
          </div>,
          "className": "subTodo-card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Drink Milk",
          },
          "ref": null,
          "rendered": "Drink Milk",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="stateButton"
            onClick={[Function]}
          >
            <FaCircleNotch
              size={18}
            />
          </button>,
          <div
            className="subTodo-card"
          >
            <div>
              Drink Milk
            </div>
          </div>,
        ],
        "className": "subTodo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaCircleNotch
              size={18}
            />,
            "className": "stateButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              Drink Milk
            </div>,
            "className": "subTodo-card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Drink Milk",
            },
            "ref": null,
            "rendered": "Drink Milk",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
